### PR TITLE
Add initial longread QC filtering - requires filtlong module update

### LINKED ISSUE #69: Read quality filtering for Nanopore data
### Description of feature
We need to add an optional complexity filtering step for Nanopore data.

### What needs to be done:
- Investigate the available tools for Nanopore and implement/add it to nf-core/taxprofiler. https://github.com/B-UMMI/long-read-catalog.
- [Filtlong](https://github.com/nf-core/modules/tree/master/modules/filtlong) is already implemented as a module.

### This can be closed when:
 - [ ] One of the tools (filtlong/nanofilt) has been merged to nf-core/taxprofiler

### PR DESCRIPTION
**Pull request by jfy133**
Closes #69 

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by jfy133** [comment]
The filtlong error:

```
Error executing process > 'NFCORE_TAXPROFILER:TAXPROFILER:LONGREAD_PREPROCESSING:FILTLONG (ERR3201952)'

Caused by:
  Process `NFCORE_TAXPROFILER:TAXPROFILER:LONGREAD_PREPROCESSING:FILTLONG (ERR3201952)` terminated with an error exit status (1)

Command executed:

  filtlong \
      -1  \
      --min_length 1000 --keep_percent 90 --target_bases 500000000 \
      ERR3201952_ERR3201952.fastq.gz \
      | gzip -n > ERR3201952_ERR3201952_lr_filtlong.fastq.gz
  
  cat <<-END_VERSIONS > versions.yml
  "NFCORE_TAXPROFILER:TAXPROFILER:LONGREAD_PREPROCESSING:FILTLONG":
      filtlong: $( filtlong --version | sed -e "s/Filtlong v//g" )
  END_VERSIONS

Command exit status:
  1

Command output:
  (empty)

Command error:
  Error: passed in argument, but no positional arguments were ready to receive it: ERR3201952_ERR3201952.fastq.gz

Work dir:
  /home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/98/e563208a0af8b1d3a74bb05ec6efe8

Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`
```

File: nextflow.config [hunk 1]
```diff
@@ -55,16 +55,23 @@ params {
...
+    shortread_qc_adapter1            = null
+    shortread_qc_adapter2            = null
+    shortread_qc_minlength           = 15
+
+    perform_longread_qc              = false
+    longread_qc_run_clip             = false
+    longread_qc_run_filter           = false
+    longread_qc_minlength            = 1000
+    longread_qc_keepbppercent        = 90
+    longread_qc_targetnbases         = 500000000
```
Comment (sofstam) [inline nextflow.config:72]: Maybe `longread_qc_target_bases` instead?

File: conf/test.config [hunk 2]
```diff
@@ -24,8 +24,8 @@ params {
     // TODO nf-core: Give any required params for the test so that command line flags are not needed
     input                                 = 'https://raw.githubusercontent.com/nf-core/test-datasets/taxprofiler/samplesheet.csv'
     databases                             = 'https://raw.githubusercontent.com/nf-core/test-datasets/taxprofiler/database.csv'
-    perform_shortread_clipmerge           = true
-    perform_longread_clip                 = false
+    perform_shortread_qc           = true
+    perform_longread_qc                 = false
```
Comment (sofstam) [inline conf/test.config:28]: Is it `false` here because it is an optional step?

Comment (jfy133) [reply conf/test.config]: Copy paste error :sweat_smile:

**Review by sofstam — APPROVED** [review]
Looks good to me once the tests pass.

**Pull request merged**